src/sources/socket/tcp.rs

```diff
@@ -33,37 +33,50 @@ pub struct TcpConfig {
     /// The maximum buffer size, in bytes, of incoming messages.
     ///
     /// Messages larger than this are truncated.
+    ///
+    /// This option is deprecated. Configure `max_length` on the framing config instead.
+    #[configurable(deprecated)]
+    #[configurable(metadata(docs::type_unit = "bytes"))]
     max_length: Option<usize>,
 
     /// The timeout, in seconds, before a connection is forcefully closed during shutdown.
     #[serde(default = "default_shutdown_timeout_secs")]
-    shutdown_timeout_secs: u64,
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    shutdown_timeout_secs: Duration,
 
     /// Overrides the name of the log field used to add the peer host to each event.
     ///
     /// The value will be the peer host's address, including the port i.e. `1.2.3.4:9000`.
     ///
     /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
     ///
+    /// Set to `""` to suppress this key.
+    ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
-    host_key: Option<OptionalValuePath>,
+    #[serde(default = "default_host_key")]
+    host_key: OptionalValuePath,
 
     /// Overrides the name of the log field used to add the peer host's port to each event.
     ///
     /// The value will be the peer host's port i.e. `9000`.
     ///
     /// By default, `"port"` is used.
-    port_key: Option<OptionalValuePath>,
+    ///
+    /// Set to `""` to suppress this key.
+    #[serde(default = "default_port_key")]
+    port_key: OptionalValuePath,
 
     #[configurable(derived)]
     tls: Option<TlsSourceConfig>,
 
     /// The size, in bytes, of the receive buffer used for each connection.
     ///
-    /// This should not typically needed to be changed.
+    /// Generally this should not need to be configured.
+    #[configurable(metadata(docs::type_unit = "bytes"))]
     receive_buffer_bytes: Option<usize>,
 
     /// The maximum number of TCP connections that will be allowed at any given time.
+    #[configurable(metadata(docs::type_unit = "concurrency"))]
```

Comment:
🤔 "concurrency" wouldn't the unit be "connections"?

Reply:
That does make more sense heh. I was actually going to omit the units altogether, but kept it for consistency.
I grepped the project for concurrency and not surprisingly there are references so I was like, maybe this was intentional. Connections does make way more sense than concurrency.

Reply:
Yeah, ran into similar in the `file` source where I made the units "lines"

Reply:
Actually, want to make sure. I found this.

```
For example, the [file source](https://vector.dev/docs/reference/configuration/sources/file/) implements concurrency across the number of files it’s tailing, and the [socket source](https://vector.dev/docs/reference/configuration/sources/socket/) implements concurrency across the number active open connection it’s maintaining.
```

https://vector.dev/docs/about/under-the-hood/architecture/concurrency-model/

Reply:
Yeah, so I read that as the `file` source implements concurrency by tailing multiple files, therefore if we had an option to control the max concurrency the unit of concurrency would be `file(s)`. Similarly `socket` does so across active multiple connections, so the max concurrency is controlled by an option with a unit of `connection(s)`

Reply:
I changed to `connections`. but there is one source, aws_sqs, which is using `concurrency` explicitly and after looking at the source code a bit I decided to keep it as is for the time being.

Reply:
That's the `Number of concurrent tasks to create for polling the queue for messages.` one right?

I think that's fine to leave as-is, I don't remember if I've done the `aws_sqs` source yet - but I can handle that regardless.

Reply:
> That's the Number of concurrent tasks to create for polling the queue for messages. one right?

Correct